src/OnboardingSPA/components/Sidebar/components/Customize/DesignColorsPanel/index.js

```diff
@@ -108,18 +119,62 @@ const DesignColorsPanel = forwardRef(
 		const [ selectedPalette, setSelectedPalette ] = useState( null );
 		const [ colorPickerCalledBy, setColorPickerCalledBy ] = useState( '' );
 		const [ showColorPicker, setShowColorPicker ] = useState( false );
-		const customPaletteId = colors.length - 1;
+
+		useEffect( () => {
+			const slug = currentData.sitegen?.homepages?.active?.slug;
+			if ( slug && ! customColors ) {
+				const storedCustomColors =
+					currentData.sitegen.homepages.data[ slug ].color
+						.customColors;
```

Comment:
Can we set the `slug` as custom instead of introducing a new key like `customColors`?